drizzlepac/align.py

```diff
@@ -483,28 +489,35 @@ def perform_align(input_list, catalog_list, num_sources, archive=False, clobber=
 
                         # Figure out which fit solution to go with based on fit_quality value and maybe also total_rms
                         if fit_quality < 5:
+                            best_matches = fit_num > best_num_matches
                             if fit_quality == 1:  # valid, non-comprimised solution with total rms < 10 mas...go with this solution.
                                 best_fit_rms = fit_rms
                                 best_fit_label = (catalog_name, algorithm_name)
+                                best_num_matches = fit_num
 
                                 best_fit_status_dict = fit_status_dict.copy()
                                 best_fit_qual = fit_quality
                                 break  # break out of while loop
-                            elif fit_quality < best_fit_qual:  # better solution found. keep looping but with the better solution as "best" for now.
+                            elif fit_quality < best_fit_qual:
+                                # better solution found. keep looping but with the better solution as "best" for now.
                                 log.info("Better solution found!")
                                 best_fit_rms = fit_rms
+                                best_num_matches = fit_num
                                 best_fit_label = (catalog_name, algorithm_name)
 
                                 best_fit_status_dict = fit_status_dict.copy()
                                 best_fit_qual = fit_quality
-                            elif fit_quality == best_fit_qual:  # new solution same level of fit_quality. Choose whichever one has the lowest total rms as "best" and keep looping.
+                            elif fit_quality == best_fit_qual:
+                                # new solution same level of fit_quality. Choose whichever one has the lowest total rms as "best" and keep looping.
```

Comment:
Update comment as the solution also needs better number of matches.

Reply:
Updated in db1f6d1.